DifferentiableHOS/statistics/l1norm.py

```diff
@@ -0,0 +1,58 @@
+# This module implements the l1 norm statistic described in
+# Ajani et al. 2021 https://arxiv.org/abs/2101.01542
+import tensorflow as tf
+import numpy as np
+from DifferentiableHOS.transforms import starlet2d
+
+__all__ = ['l1norm']
+
+
+def _get_wavelet_normalization(image, nscales):
+  """ Computes normalizing constant for starlet, for given image.
+  """
+  _, nx, ny = image.get_shape()
+  knorm = tf.ones((1, 1, 1, 1), dtype=tf.float32)
+  knorm = tf.image.resize_with_crop_or_pad(knorm, nx, ny)
+  wt = starlet2d(knorm[..., 0], nscales=nscales)
+  return [tf.math.sqrt(tf.reduce_sum(c**2)) for c in wt]
+
+
+def l1norm(image, nscales=3, value_range=[-1., 1.], nbins=16, name='l1norm'):
```

Comment:
is this one the differentiable implementation of the  l1norm?

Reply:
hum no :-( sorry I forgot to push it last night,